urx/ursecmon.py

```diff
@@ -340,7 +340,7 @@ def _get_data(self):
                 tmp = self._s_secondary.recv(1024)
                 self._dataqueue += tmp
 
-    def wait(self, timeout=0.5):
+    def wait(self, timeout=2.0):
```

Comment:
I see you increased everywhere the timeouts. These timeouts were not random. if it takes you more than 0.5s to create a connection then you have an issue in your network

Reply:
examining the issues and forks, I could see that several people had these timeout errors when creating the connection, which were easily "resolved" by increasing the timeout to become usable